commons/all/src/main/java/org/infinispan/commons/internal/CommonsBlockHoundIntegration.java

```diff
@@ -59,5 +65,28 @@ private static void handleJREClasses(BlockHound.Builder builder) {
       builder.allowBlockingCallsInside(ForkJoinPool.class.getName(), "runWorker");
       // The scan method is where the task is actually ran
       builder.disallowBlockingCallsInside(ForkJoinPool.class.getName(), "scan");
+
+      // SecureRandom reads from a socket
+      builder.allowBlockingCallsInside(SecureRandom.class.getName(), "nextBytes");
+
+      // Just assume all the thread pools don't block - NOTE rejection policy can still be an issue!
+      allowMethodsToBlock(builder, ThreadPoolExecutor.class, true);
+      allowMethodsToBlock(builder, ScheduledThreadPoolExecutor.class, true);
+      builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "getTask");
+      builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "processWorkerExit");
+
+      // Allow logging to block
+      builder.allowBlockingCallsInside(java.util.logging.Logger.class.getName(), "log");
+   }
+
+   private static void log4j(BlockHound.Builder builder) {
+      try {
+         Class.forName("org.apache.logging.log4j.spi.AbstractLogger");
+         builder.allowBlockingCallsInside(AbstractLogger.class.getName(), "logMessage");
+      } catch (ClassNotFoundException e) {
+         throw new RuntimeException(e);
```

Comment:
Should it fail without the AbstractLogger?

Reply:
Oh whoops, let me fix that.